src/vma/sock/sockinfo_tcp.cpp

```diff
@@ -3681,6 +3690,23 @@ int sockinfo_tcp::setsockopt(int __level, int __optname,
 			unlock_tcp_con();
 			si_tcp_logdbg("(TCP_QUICKACK) value: %d", val);
 			break;
+		case TCP_KEEPIDLE: {
+			unsigned int idle_sec = *(unsigned int *)__optval;
+			si_tcp_logdbg("TCP_KEEPIDLE value: %us", idle_sec);
+			m_pcb.keep_idle = idle_sec * 1000U;
+		} break;
+#if LWIP_TCP_KEEPALIVE
+		case TCP_KEEPINTVL: {
```

Comment:
add checks for valid values.
see kernel for example:
	if (val < 1 || val > MAX_TCP_KEEPINTVL)
		return -EINVAL;

Reply:
I agree that the value must be positive and even greater than 0, but [MAX_TCP_KEEPIDLE](https://elixir.bootlin.com/linux/latest/C/ident/MAX_TCP_KEEPIDLE)/[MAX_TCP_KEEPINTVL](https://elixir.bootlin.com/linux/latest/C/ident/MAX_TCP_KEEPINTVL)/[MAX_TCP_KEEPCNT](https://elixir.bootlin.com/linux/latest/C/ident/MAX_TCP_KEEPCNT) are there to ensure boundary checks relevant for Linux internal implementation. Since LwIP uses u_32t for three of the arguments and since the parameter is int the upper-bound check is irrelevant.

Reply:
I agree that we should comply with the Linux boundaries.